input-output-hk/daedalus
# [DDW-535] Support frontend only mode

This PR adds support for `frontendOnlyMode` which makes it possible to connect to manually started instances of cardano-node for advanced debugging purposes.

## How to Test:
1. Within the cardano-sl repository you first have to build a script for a certain network … e.g for testnet: `nix-build -A connectScripts.testnet.wallet -o launch_testnet`
2. Then launch this cluster + node with `./launch_testnet`
3. You should now have a `state-wallet-testnet` folder inside the cardano-sl repo … copy the full path to the sub folder `tls` in there.
4. Within the Daedalus repo checkout this branch and run: `CARDANO_TLS_PATH=/path/to/tls CARDANO_HOST=localhost CARDANO_PORT=8090 nix-shell`

Now you should have a pre-configured nix-shell session where you can `yarn dev` as usual and Daedalus connects itself to the manually started cardano node.

## Note:

Only `CARDANO_TLS_PATH` is mandatory. `CARDANO_HOST` defaults to `localhost` and `CARDANO_PORT` to 8090 😉 

So if you just start the default cardano node (which runs on localhost:8090) you can also start nix-shell with `CARDANO_TLS_PATH=/path/to/tls nix-shell`

# Todos:

- [x] The `frontendOnlyMode` flag in `launcherConfig` is always `true` … it should be false when `--arg autoStartBackend true` is passed to the nix-shell command. This means that currently only the manual cardano node mode is working (since it's always assuming that it should run in frontend only mode!)
- [x] Update Electron version used by `nix-shell` (@cleverca22) - fixed in https://github.com/input-output-hk/daedalus/pull/1241/commits/045fc7ac814b24bdee2592c0bf7805ce93d055c9

---

## Review Checklist:

### Basics


### Code Quality

### Testing

### After Review:
